sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/security/CertificateNamesGenerator.java

```diff
@@ -41,15 +42,19 @@
     // ub-common-name INTEGER ::= 64
     private static final int CN_MAX_LENGTH = 64;
 
-    public CertificateNamesGenerator(String serviceName, TaskSpec taskSpec, PodInstance podInstance) {
+    public CertificateNamesGenerator(String serviceName,
+                                     TaskSpec taskSpec,
+                                     PodInstance podInstance,
+                                     SchedulerConfig schedulerConfig) {
         this.serviceName = serviceName;
         this.taskInstanceName = TaskSpec.getInstanceName(podInstance, taskSpec);
         // Task can specify its own service discovery name
         if (taskSpec.getDiscovery().isPresent() && taskSpec.getDiscovery().get().getPrefix().isPresent()) {
```

Comment:
This seems to indicate that we already have some form of discovery in the `taskSpec`. Could we not use that for the custom TLD too?

Reply:
Also, while we're making the change, could we use:
```java
        String taskName = taskSpec.getDiscovery().flatMap(d -> d.getPrefix())
                .map(prefix -> String.format("%s-%d", prefix, podInstance.getIndex()))
                .orElse(this.taskInstanceName);
        this.autoIpHostname = EndpointUtils.toAutoIpHostname(serviceName, taskName, schedulerConfig);
```

Reply:
That would require first classing this in the YAML.

I am hesitant to do so. Leaving it as a sneaky envvar keeps it secreter.